paasta_tools/cli/cmds/check.py

```diff
@@ -116,19 +116,18 @@ def docker_check():
 
 
 def makefile_responds_to(target):
-    """Runs `make -q <target>` to detect if a makefile responds to the
+    """Runs `make --dry-run <target>` to detect if a makefile responds to the
     specified target."""
-    cmd = 'make -q %s' % target
+    # `make -q` is commonly used to detect whether a Makefile responds to a specified target.
```

Comment:
I wouldn't bother with all these comments debating why not use -q, I think --dry-run is obvious enough